src/main/java/org/opensearch/geospatial/geojson/FeatureCollection.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.geojson;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import org.opensearch.geospatial.GeospatialParser;
+
+/**
+ * FeatureCollection represents GEOJSON of type FeatureCollection. A FeatureCollection object has a member
+ * with the name "features".  The value of "features" is a List.
+ * Each element of the list is a {@link Feature} object.  It
+ * is possible for this list to be empty.
+ */
+public final class FeatureCollection {
+    public static final String TYPE = "FeatureCollection";
+    public static final String FEATURES_KEY = "features";
+    public static final String TYPE_KEY = "type";
+    private final List<Map<String, Object>> features;
```

Comment:
Why isnt this a list of Features?

Reply:
The strategy is , user will pass input as an array of Map. This input will be GeoJSON and it can be either Feature or FeatureCollection. From, API POV, it only needs Feature. So, if user passes FeatureCollection, we only need Map of Features. Hence, there is no need to convert this map to Feature.

Reply:
Oops. I had that as Feature and later when i started using this in next PR, i realized i don't need to convert, but forgot to update documentation. Corrected now.